### ISSUE TITLE
get_helm.sh wrong usage of openssl

### ISSUE THREAD

**Issue by joestack** [kubernetes/helm#2443]
for me it seems there is a syntax error in "get_helm.sh"

line 122
-  local sum=$(openssl sha -sha256 ${HELM_TMP_FILE} | awk '{print $2}')
+ local sum=$(openssl sha1 -sha256 ${HELM_TMP_FILE} | awk '{print $2}')

**Comment by jascott1** [comment]
@joestack Are you saying you don't have `sha` in your installed openssl?

**Comment by joestack** [comment]
I might do something wrong, but my openssl offers "sha1" only and not "sha".
Changing the value within the get_helm.sh and the script works fine on 
Debian 8.0.


#1#
curl 
https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get > 
get_helm.sh
   % Total    % Received % Xferd  Average Speed   Time    Time     Time 
Current
                                  Dload  Upload   Total   Spent    Left 
Speed
100  5962  100  5962    0     0  24430      0 --:--:-- --:--:-- --:--:-- 
24534

#2#
chmod 700 get_helm.sh

#3#
./get_helm.sh
openssl sha -sha256+  /tmp/helm-v2.4.1-linux-amd64.tar.gzawk {print $2}

Invalid command 'sha'; type "help" for a list.

#4#
openssl help

Standard commands
asn1parse         ca                ciphers           cms
crl               crl2pkcs7         dgst              dhparam
dsa               dsaparam          ec                ecparam
enc               engine            errstr            exit
gendsa            genpkey           genrsa            help
list              nseq              ocsp              passwd
pkcs12            pkcs7             pkcs8             pkey
pkeyparam         pkeyutl           prime             rand
rehash            req               rsa               rsautl
s_client          s_server          s_time            sess_id
smime             speed             spkac             srp
ts                verify            version           x509

Message Digest commands (see the `dgst' command for more details)
blake2b512        blake2s256        gost              md4
md5               rmd160            sha1              sha224
sha256            sha384            sha512

Cipher commands (see the `enc' command for more details)
aes-128-cbc       aes-128-ecb       aes-192-cbc       aes-192-ecb
aes-256-cbc       aes-256-ecb       base64            bf
bf-cbc            bf-cfb            bf-ecb            bf-ofb
camellia-128-cbc  camellia-128-ecb  camellia-192-cbc  camellia-192-ecb
camellia-256-cbc  camellia-256-ecb  cast              cast-cbc
cast5-cbc         cast5-cfb         cast5-ecb         cast5-ofb
des               des-cbc           des-cfb           des-ecb
des-ede           des-ede-cbc       des-ede-cfb       des-ede-ofb
des-ede3          des-ede3-cbc      des-ede3-cfb      des-ede3-ofb
des-ofb           des3              desx              rc2
rc2-40-cbc        rc2-64-cbc        rc2-cbc           rc2-cfb
rc2-ecb           rc2-ofb           rc4               rc4-40
seed              seed-cbc          seed-cfb          seed-ecb
seed-ofb

#5#
dpkg -la | grep openssl
ii  openssl                                                     1.1.0e-1 
                             amd64        Secure Sockets Layer toolkit - 
cryptographic utility
ii  perl-openssl-defaults:amd64                                 3 
                             amd64        version compatibility baseline 
for Perl OpenSSL packages
ii  python-openssl                                              16.2.0-1 
                             all          Python 2 wrapper around the 
OpenSSL library

**Comment by technosophos** [comment]
It's unclear to me if you have found a fix. Does it work as expected now?

**Comment by joestack** [comment]
Yes. In Case of using sha1 instead of sha, the sum is identical. 


Von meinem Samsung Galaxy Smartphone gesendet.
-------- Ursprüngliche Nachricht --------Von: Matt Butcher <notifications@github.com> Datum: 16.05.17  18:31  (GMT+01:00) An: kubernetes/helm <helm@noreply.github.com> Cc: JoeStack <joern@tel2ip.net>, Mention <mention@noreply.github.com> Betreff: Re: [kubernetes/helm] get_helm.sh wrong usage of openssl (#2443) 
It's unclear to me if you have found a fix. Does it work as expected now?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/kubernetes/helm","title":"kubernetes/helm","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/kubernetes/helm"}},"updates":{"snippets":[{"icon":"PERSON","message":"@technosophos in #2443: It's unclear to me if you have found a fix. Does it work as expected now?"}],"action":{"name":"View Issue","url":"https://github.com/kubernetes/helm/issues/2443#issuecomment-301838516"}}}

**Comment by technosophos** [comment]
marking resolved

**Comment by mtricolici** [comment]
why this issue has been closed? What's the fix for this?!

**Comment by mtricolici** [comment]
```
$cat Dockerfile 
FROM debian:9

RUN apt-get update -q && \
  apt-get install -yq curl openssl && \
  curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | bash

$docker build -t zzz .
...
SHA sum of /tmp/helm does not match. Aborting.
```

**Comment by jascott1** [comment]
@mtricolici The workaround is to edit the `get_helm.sh` as @joestack pointed out:

>my openssl offers "sha1" only and not "sha".
>Changing the value within the get_helm.sh and the script works fine on
>Debian 8.0.

So some OS's openssl have `sha` and others  `sha1`. The get_helm.sh script uses `sha`.
To workaround if you only have `sha1` edit get_helm.sh and change `openssl sha` to `openssl sha1`

Also please know that you can just manually download the Helm release and copy the `helm` binary to somewhere accessible from your PATH. After that you run `helm init`.

**Comment by ljnelson** [comment]
Old issue but just pipe it through `sed` 😄 :

    curl https://raw.githubusercontent.com/kubernetes/helm/master/scripts/get | sed 's/openssl sha /openssl sha1 /g' | bash

**Comment by jascott1** [comment]
@ljnelson FYI We replaced `sha` with `sha1` since the popular distros all support it. This was fixed in master but didn't get into 2.6.1. I expect it to be in next release.
https://github.com/kubernetes/helm/pull/2860
